### PR TITLE
Fix: "Generate an error if a user wrongly inputs options with –" 

### LINKED ISSUE #484: NSFS: generic bug: noobaa install not reporting erroneous input for options
This issue is tagged with NSFS but bug reported here is generic.

currently noobaa install does not report any errors if a user wrongly inputs options with  – vs. --. Ex: When user wrongly inputs –pv-pool-default-storage-class='<storage class>' instead of --pv-pool-default-storage-class='<storage class>', the install proceeds without reporting any errors and completes. But after the install `noobaa-default-backing-store` pod remains in Pending state showing there is no storage provision-er detected for pvc. 

If user has used wrong syntax unknowingly then there is lot of time spent on seeing why the pvc was not able to create before tracing it back to the incorrect syntax used.

```
# noobaa install --image-pull-secret='myregistrykey' --db-storage-class='<storage class>' –pv-pool-default-storage-class='<storage class>'
INFO[0000] CLI version: 5.5.0-nsfs                      
INFO[0000] noobaa-image: noobaa/noobaa-core:5.5.0-nsfs  
INFO[0000] operator-image: noobaa/noobaa-operator:5.5.0-nsfs 
INFO[0000] Namespace: noobaa                            
INFO[0000]                                              
INFO[0000] CRD Create:                                  
INFO[0000] ✅ Created: CustomResourceDefinition "noobaas.noobaa.io" 
INFO[0000] ✅ Created: CustomResourceDefinition "backingstores.noobaa.io" 
INFO[0000] ✅ Created: CustomResourceDefinition "bucketclasses.noobaa.io" 
INFO[0000] ✅ Created: CustomResourceDefinition "objectbucketclaims.objectbucket.io" 
INFO[0000] ✅ Created: CustomResourceDefinition "objectbuckets.objectbucket.io" 
INFO[0000]                                              
INFO[0000] Operator Install:                            
INFO[0000] ✅ Already Exists: Namespace "noobaa"         
INFO[0000] ✅ Created: ServiceAccount "noobaa"           
INFO[0000] ✅ Created: Role "noobaa"                     
INFO[0000] ✅ Created: RoleBinding "noobaa"              
INFO[0000] ✅ Created: ClusterRole "noobaa.noobaa.io"    
INFO[0000] ✅ Created: ClusterRoleBinding "noobaa.noobaa.io" 
INFO[0000] ✅ Created: Deployment "noobaa-operator"      
INFO[0000]                                              
INFO[0000] System Create:                               
INFO[0000] ✅ Already Exists: Namespace "noobaa"         
INFO[0002] ✅ Created: NooBaa "noobaa"                   
INFO[0002]                                              
INFO[0002] NOTE:                                        
INFO[0002]   - This command has finished applying changes to the cluster. 
INFO[0002]   - From now on, it only loops and reads the status, to monitor the operator work. 
INFO[0002]   - You may Ctrl-C at any time to stop the loop and watch it manually. 
INFO[0002]                                              
INFO[0002] System Wait Ready:                           
INFO[0002] ⏳ System Phase is "". Pod "noobaa-operator-888f55c89-jfkbb" is not yet ready: Phase="Pending". ContainersNotReady (containers with unready status: [noobaa-operator]). ContainersNotReady (containers with unready status: [noobaa-operator]).  
INFO[0005] ⏳ System Phase is "". Pod "noobaa-operator-888f55c89-jfkbb" is not yet ready: Phase="Pending". ContainersNotReady (containers with unready status: [noobaa-operator]). ContainersNotReady (containers with unready status: [noobaa-operator]).  
INFO[0008] ⏳ System Phase is "". Pod "noobaa-operator-888f55c89-jfkbb" is not yet ready: Phase="Pending". ContainersNotReady (containers with unready status: [noobaa-operator]). ContainersNotReady (containers with unready status: [noobaa-operator]).  
INFO[0011] ⏳ System Phase is "". Pod "noobaa-operator-888f55c89-jfkbb" is not yet ready: Phase="Pending". ContainersNotReady (containers with unready status: [noobaa-operator]). ContainersNotReady (containers with unready status: [noobaa-operator]).  
INFO[0014] ⏳ System Phase is "". StatefulSet "noobaa-core" is not found yet 
INFO[0017] ⏳ System Phase is "". StatefulSet "noobaa-core" is not found yet 
INFO[0020] ⏳ System Phase is "". StatefulSet "noobaa-core" is not found yet 
INFO[0023] ⏳ System Phase is "". Pod "noobaa-core-0" is not yet ready: Phase="Pending". ContainersNotReady (containers with unready status: [core]). ContainersNotReady (containers with unready status: [core]).  
INFO[0026] ⏳ System Phase is "". Pod "noobaa-core-0" is not yet ready: Phase="Pending". ContainersNotReady (containers with unready status: [core]). ContainersNotReady (containers with unready status: [core]).  
..
INFO[0134] ✅ Exists: PersistentVolumeClaim "db-noobaa-db-0" 
INFO[0134] ✅ System Phase is "Ready"                    
INFO[0134] ✅ Exists:  "noobaa-admin"                    
..
#------------------#
#- Backing Stores -#
#------------------#

NAME                           TYPE      TARGET-BUCKET   PHASE      AGE   
noobaa-default-backing-store   pv-pool                   Creating   13s   

#------------------#
#- Bucket Classes -#
#------------------#

NAME                          PLACEMENT                                                             PHASE       AGE   
noobaa-default-bucket-class   {Tiers:[{Placement: BackingStores:[noobaa-default-backing-store]}]}   Verifying   22s   

#-----------------#
#- Bucket Claims -#
#-----------------#

No OBCs found.

[root@fyreauto-x-master backup]# kubectl get pvc
NAME                                               STATUS    VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                     AGE
db-noobaa-db-0                                     Bound     pvc-527e95c6-68ee-47c3-a652-e26ce14fc7eb   50Gi       RWO            <storage class>   2m24s
noobaa-default-backing-store-noobaa-pvc-a3e29bef   Pending                                                                                                         41s
```

Expected behavior: noobaa installer should report the incorrect syntax and exit.

### PR DESCRIPTION
**Pull request by baum**
### Issue

Fixes #484
 
### Description

Use cobra args validations - https://github.com/spf13/cobra#positional-and-custom-arguments

Signed-off-by: Alexander Indenbaum <aindenba@redhat.com>

### CODE REVIEW

**Comment by liranmauda** [comment]
@baum I think as this PR reverting a commit, it is better to keep at list 2 commits here.
1 for the revert and the other for the change.

File: pkg/backingstore/backingstore.go [hunk 1]
```diff
@@ -824,8 +821,8 @@ func RunReconcile(cmd *cobra.Command, args []string) {
 	}
 	backingStoreName := args[0]
 	klient := util.KubeClient()
-	intervalDuration := time.Duration(3)
-	util.Panic(wait.PollImmediateInfinite(intervalDuration*time.Second, func() (bool, error) {
+	intervalSec := time.Duration(3)
```
Comment (liranmauda) [inline pkg/backingstore/backingstore.go:824]: Referring to Guy's suggestion on the previous PR:
https://github.com/noobaa/noobaa-operator/pull/646#discussion_r651707785

File: pkg/bucketclass/bucketclass.go [hunk 2]
```diff
@@ -497,8 +495,8 @@ func RunReconcile(cmd *cobra.Command, args []string) {
 	}
 	bucketClassName := args[0]
 	klient := util.KubeClient()
-	intervalDuration := time.Duration(3)
-	util.Panic(wait.PollImmediateInfinite(intervalDuration*time.Second, func() (bool, error) {
+	intervalSec := time.Duration(3)
```
Comment (liranmauda) [inline pkg/bucketclass/bucketclass.go:498]: Referring to Guy's suggestion on the previous PR:
https://github.com/noobaa/noobaa-operator/pull/646#discussion_r651707785

File: pkg/namespacestore/namespacestore.go [hunk 3]
```diff
@@ -632,8 +629,8 @@ func RunReconcile(cmd *cobra.Command, args []string) {
 	}
 	namespaceStoreName := args[0]
 	klient := util.KubeClient()
-	intervalDuration := time.Duration(3)
-	util.Panic(wait.PollImmediateInfinite(intervalDuration*time.Second, func() (bool, error) {
+	intervalSec := time.Duration(3)
```
Comment (liranmauda) [inline pkg/namespacestore/namespacestore.go:632]: Referring to Guy's suggestion on the previous PR:
https://github.com/noobaa/noobaa-operator/pull/646#discussion_r651707785

File: pkg/backingstore/backingstore.go (hunk 1 above)
Comment (baum) [reply pkg/backingstore/backingstore.go:824]: Just left this change alone, unrelated to the issue. Reverted to the original.

File: pkg/bucketclass/bucketclass.go (hunk 2 above)
Comment (baum) [reply pkg/bucketclass/bucketclass.go:498]: Reverted to the original, unrelated to the current issues.

File: pkg/namespacestore/namespacestore.go (hunk 3 above)
Comment (baum) [reply pkg/namespacestore/namespacestore.go:632]: Reverted to the original, unrelated to the current issues.

**Comment by dannyzaken** [comment]
after discussing with @baum, he will separate this PR into 2 PRs. 
* this PR to revert #646, so we can merge it quickly and fix the regression. 
* a new PR with a new fix for #484 so we can test it better before merging

File: pkg/backingstore/backingstore.go [hunk 4]
```diff
@@ -273,6 +274,7 @@ func CmdReconcile() *cobra.Command {
 		Use:    "reconcile",
 		Short:  "Runs a reconcile attempt like noobaa-operator",
 		Run:    RunReconcile,
+		Args:   cobra.NoArgs,
```
Comment (dannyzaken) [inline pkg/backingstore/backingstore.go:277]: reconcile expects args. 
https://github.com/baum/noobaa-operator/blob/c88f9a3c614a95083bce17e6539e2b2c2de67671/pkg/backingstore/backingstore.go#L819-L819

File: pkg/bucketclass/bucketclass.go [hunk 5]
```diff
@@ -176,6 +177,7 @@ func CmdReconcile() *cobra.Command {
 		Use:    "reconcile",
 		Short:  "Runs a reconcile attempt like noobaa-operator",
 		Run:    RunReconcile,
```
Comment (dannyzaken) [inline pkg/bucketclass/bucketclass.go:179]: same as for backingstore - reconcile expects args

**Review by dannyzaken — CHANGES_REQUESTED** [review]

File: pkg/backingstore/backingstore.go (hunk 4 above)
Comment (baum) [reply pkg/backingstore/backingstore.go]: 10x @dannyzaken, removed.

File: pkg/bucketclass/bucketclass.go (hunk 5 above)
Comment (baum) [reply pkg/bucketclass/bucketclass.go]: 10x

**Review by dannyzaken — APPROVED** [review]

**Pull request merged**